wps-common/src/main/java/au/org/aodn/aws/wps/status/ExecuteStatusBuilder.java

```diff
@@ -36,71 +26,77 @@ public String getStatusLocation() {
+     *
+     * @param jobStatus
+     * @param failedMessage
+     * @param failedCode
+     * @param outputs
+     * @return
+     */
+    public String createResponseDocument(EnumStatus jobStatus, String failedMessage, String failedCode, HashMap<String, String> outputs) {
 
         ExecuteResponse response = new ExecuteResponse();
         response.setStatusLocation(getStatusLocation());
 
         StatusType status = new StatusType();
 
-        try {
-            GregorianCalendar currentTime = new GregorianCalendar();
-            currentTime.setTime(new Date());
-            currentTime.setTimeZone(TimeZone.getTimeZone("GMT"));
-            XMLGregorianCalendar xmlDate = DatatypeFactory.newInstance().newXMLGregorianCalendar(currentTime);
-
-            status.setCreationTime(xmlDate);
-        } catch (DatatypeConfigurationException e) {
+        try
+        {
+            status.setCreationTime(StatusHelper.getCreationDate());
+        }
+        catch (DatatypeConfigurationException e)
+        {
             throw new RuntimeException(e);
         }
 
-        if (jobStatus==EnumStatus.ACCEPTED) {
+        if (jobStatus==EnumStatus.ACCEPTED)
+        {
             status.setProcessAccepted("Accepted job " + jobId + " for processing");
-        } else if (jobStatus==EnumStatus.STARTED) {
-            ProcessStartedType started = new ProcessStartedType();
-            started.setValue("Job " + jobId + " is currently running");
-            status.setProcessStarted(started);
-        } else if (jobStatus==EnumStatus.PAUSED) {
-            ProcessStartedType started = new ProcessStartedType();
-            started.setValue("Job " + jobId + " is currently paused");
-            status.setProcessPaused(started);
-        }  else if (jobStatus==EnumStatus.SUCCEEDED) {
+        }
+        else if (jobStatus==EnumStatus.STARTED)
+        {
+            status.setProcessStarted(StatusHelper.getProcessStartedType("Job " + jobId + " is currently running", new Integer(0)));
+        }
+        else if (jobStatus==EnumStatus.PAUSED)
+        {
+            status.setProcessPaused(StatusHelper.getProcessStartedType("Job " + jobId + " is currently paused", new Integer(0)));
+        }
+        else if (jobStatus==EnumStatus.SUCCEEDED)
+        {
             status.setProcessSucceeded("Job " + jobId + " has completed");
-        } else if (jobStatus==EnumStatus.FAILED) {
-            ProcessFailedType failed = new ProcessFailedType();
-            ExceptionReport report = new ExceptionReport();
-            ExceptionType type = new ExceptionType();
-            type.getExceptionText().add(failedMessage);
-            type.setExceptionCode(failedCode);
-            report.getException().add(type);
-            failed.setExceptionReport(report);
+            //  If outputs were passed - add them to the response
+            if(outputs != null)
+            {
+                for(String currentKey : outputs.keySet())
```

Comment:
According to the interwebs this is inefficient and you should iterate via the entrySet() method if you want both key and value. Might be more readable as well?

Reply:
Changed to use entrySet()